### DISCUSSION TITLE
512 Tiles

### DISCUSSION THREAD

**Discussion by scadergit** [ghettovoice/vuelayers#444] [Q&A]
It seems to make OL request 512 tiles, you have to pass an ol/tileGrid object in the ol/TileWMS source constructor. There doesn't seem to be a tile grid component in VueLayers, so do I have to import the OL tileGrid object in my component and pass a function factory to the VL wms source component?

Actually, I was looking at master, and in 0.11.28 there is a property for tileSize...

**Comment by ghettovoice** [comment]
Hello @scadergit ,
there how it can be done:
1. in v0.11.x all tile source components have a prop `tileSize`. Pass an array `[512, 512]`.
2. in v0.12.x WMS source components have a prop `tileGridFactory`, you should pass a custom grid factory.
For example with XYZ grid:

```html
<template>
  <vl-map>
    <vl-layer-tile>
      <vl-source-tile-wms :tile-grid-factory="customGridFactory" />
    </vl-layer-tile>
  </vl-map>
</template>

<script>
import { createXYZ } from 'ol/tilegrid'

export default {
  methods: {
    customGridFactory () {
      return createXYZ({
        tileSize: [512, 512],
      })
    },
  },
}
</script>
```

**Reply by scadergit** [reply]
Thanks for verifying and the examples. Every time I read up on custom tile sizes I get lost in the resolutions rabbit hole. For now, it seems to work fine just using tileSize prop, but I wish I had a better understanding of that concept.

**Reply by ghettovoice** [reply]
You can check also the documentation and sources of `ol/tilegrid/createXYZ` here https://openlayers.org/en/latest/apidoc/module-ol_tilegrid.html. 
By default it calculates resolutions based on EPSG: 3857 proj.
